text/0026-rest-api/rest-api.yaml

```diff
@@ -0,0 +1,1777 @@
+        data:
+          type: object
+          properties:
+            messageId:
+              type: string
+              description: The message identifier of the submitted message.
+
+    MessagesFindResponse:
+      description: Searches for messages matching a given indexation key.
+      properties:
+        data:
+          type: object
+          properties:
+            index:
+              type: string
+              description: The provided indexation key that was used to search for.
+            maxResults:
+              type: integer
+              description: The number of results it can return at most.
+            count:
+              type: integer
+              description: The actual number of found results.
+            messageIds:
+              type: array
+              items:
+                type: string
+              description: The identifiers of the found messages that match the given indexation key.
+
+    MessageMetadataResponse:
+      description: Returns the metadata of a given message.
+      properties:
+        data:
+          type: object
+          properties:
+            messageId:
+              type: string
+              description: The identifier of the message.
+            parent1MessageId:
+              type: string
+              description: The identifier of the first Message it references.
+            parent2MessageId:
+              type: string
+              description: The identifier of the second Message it references.
+            isSolid:
+              type: boolean
+              description: Tells if the message could get solidified by the node or not.
+            referencedByMilestoneIndex:
+              type: integer
+              nullable: true
+              description: Tells which milestone references this message. If `null` the message was not referenced by a milestone yet.
+            milestoneIndex:
+              type: integer
+              description: If set, this message can be considered as a valid milestone message. This field therefore describes the milestone index of the involved milestone. A message can be considered as a valid milestone message if the milestone payload is valid and if the referenced parents in the milestone payload do match the referenced parents in the message itself. Note it's possible to have different milestone messages that all represent the same milestone.
+            ledgerInclusionState:
+              type: string
+              enum:
+                - included
+                - conflicting
+                - noTransaction
+              description: If `included`, the message contains a transaction that has been included in the ledger. If `conflicitng`, the message contains a transaction that has not been included in the ledger because it conflicts with another transaction. If the message does not contain a transaction, `ledgerInclusionState` is set to `noTransaction`.
+            conflictReason:
```

Comment:
Maybe document what the values can be
https://github.com/gohornet/hornet/blob/a153e109ca6517e14d8153a1e72886f025667f28/pkg/model/storage/message_metadata.go#L26-L52